### PR TITLE
feat(speech): add Chinese STT/TTS model support

### PR DESCRIPTION
**Pull request by shouwangzhe**
## Summary

Adds local Chinese speech recognition and synthesis support to Paseo via sherpa-onnx models.

## New Models

| Model ID | Type | Description |
|----------|------|-------------|
| `paraformer-zh-2024-03-09-int8` | STT | Paraformer Chinese+English (Mandarin + dialects, int8, ~217MB) |
| `sense-voice-zh-en-ja-ko-yue-int8` | STT | SenseVoice (zh/en/ja/ko/yue, auto-detected) |
| `kokoro-multi-lang-v1_0` | TTS | Kokoro multilingual (Chinese + English) |

## Code Changes

- **model-catalog.ts**: Added 3 new model entries + `sttModelType` / `supportedLanguages` fields to all existing entries + `resolveDefaultSttModelForLanguage()` / `resolveDefaultTtsModelForLanguage()` functions
- **sherpa-offline-recognizer.ts**: Extended `SherpaOfflineRecognizerModel` union type to support `sense_voice` and `paraformer` model kinds with proper sherpa-onnx config
- **worker-process.ts**: Added `buildSttModelConfig()` to dynamically build model config based on `sttModelType`
- **config.ts**: Updated default model resolution to be language-aware
- **sherpa-tts.ts**: Added `kokoro-multi-lang-v1_0` preset + lexicon file detection for multilingual TTS

## Usage

```bash
# Explicit model selection
PASEO_VOICE_LOCAL_STT_MODEL=paraformer-zh-2024-03-09-int8

# Or set language (auto-selects appropriate models)
PASEO_VOICE_LANGUAGE=zh
PASEO_DICTATION_LANGUAGE=zh
```

## Testing

- Verified Paraformer STT on standard Mandarin, mixed Chinese-English, Sichuan, Tianjin, and Henan dialect audio samples
- All existing English models unaffected (defaults unchanged)
- Model auto-downloads on first use (~217MB for Paraformer int8)

## Non-breaking

- No changes to default behavior (English STT still uses parakeet)
- No protocol or API changes
- New models are opt-in via env var or language setting

### CODE REVIEW

File: packages/server/src/server/speech/providers/local/sherpa/model-catalog.ts [hunk 1]
```diff
@@ -118,3 +173,48 @@ export function getSherpaOnnxModelSpec(id: SherpaOnnxModelId): SherpaOnnxModelSp
...
+
+/** Returns the catalog-defined default speaker ID for a TTS model, or undefined if none. */
+export function getDefaultSpeakerId(id: LocalTtsModelId): number | undefined {
+  return SHERPA_ONNX_MODEL_CATALOG[id].defaultSpeakerId as number | undefined;
+}
+
+export function resolveDefaultSttModelForLanguage(language: string): LocalSttModelId {
+  // If the global default model supports this language, keep it.
+  const defaultSpec = SHERPA_ONNX_MODEL_CATALOG[DEFAULT_LOCAL_STT_MODEL];
+  if ((defaultSpec.supportedLanguages as readonly string[] | undefined)?.includes(language)) {
```
Comment (LauraGPT) [inline packages/server/src/server/speech/providers/local/sherpa/model-catalog.ts]: Please normalize the language tag before catalog matching and use the normalized value in both STT and TTS resolvers. `LanguageSchema` accepts any non-empty string, while common configured values are BCP 47 tags such as `zh-CN` and `pt-BR`. With the current exact `includes(language)` checks, `PASEO_DICTATION_LANGUAGE=zh-CN` falls back to the English Parakeet model instead of SenseVoice, and `pt-BR` misses the multilingual Parakeet model. Lowercasing and matching the base subtag (for example `zh-CN` -> `zh`) would fix this; please cover regional tags and case variants in the focused tests.

**Review by LauraGPT — COMMENTED** [review]
Thanks for adding SenseVoice, Paraformer, and multilingual Kokoro support. The sherpa-onnx runtime config shapes match the upstream examples. I found one model-selection edge case below. Before merge, please also add focused tests for Chinese default resolution (STT -> SenseVoice, TTS -> multilingual Kokoro), explicit model overrides, unchanged English defaults, and SenseVoice/Paraformer config construction. This PR currently adds no tests and its only check is Greptile. Finally, please update `docs/data-model.md` (currently line 204), which still says local model IDs are intentionally limited to English Parakeet and Kokoro.

File: packages/server/src/server/speech/providers/local/sherpa/model-catalog.ts (hunk 1 above)
Comment (LauraGPT) [reply packages/server/src/server/speech/providers/local/sherpa/model-catalog.ts]: The BCP 47 normalization itself is fixed on the current f30b2cc head. I kept this thread open because the regional/case behavior and Chinese defaults still had no regression coverage in the source branch. I opened [shouwangzhe/paseo#1](https://github.com/shouwangzhe/paseo/pull/1) against this exact branch with the requested language/default/override tests, native SenseVoice and Paraformer config tests, and the model-ID docs update. The focused suite is 16/16 green and the full server typecheck, Oxfmt, Oxlint, and diff checks pass. Once that companion is merged, this thread can be resolved.

Comment (LauraGPT) [reply packages/server/src/server/speech/providers/local/sherpa/model-catalog.ts]: Confirmed on current upstream head `d3b1bdae`: the BCP 47 normalization, regional/case regressions, Chinese default/override coverage, native SenseVoice/Paraformer config tests, and model-ID docs update are all present after merging `shouwangzhe/paseo#1`.

Fresh current-head verification passes 21 focused speech tests, the full server typecheck, and changed-file formatting. Resolving this original language-selection thread as addressed.

**Review by LauraGPT — COMMENTED** [review]
Current-head validation for `d3b1bdae` found the earlier language-selection requirements addressed, but the changed-file lint gate still has two contributor-side failures:

1. `packages/server/src/server/speech/providers/local/sherpa/sherpa-offline-recognizer.ts:64`: `eslint(complexity)` reports constructor complexity **26**, above the repository limit of 20. The model-kind switch/config construction should move into a focused helper so the constructor returns below the enforced threshold.
2. `packages/server/src/server/speech/providers/local/sherpa/sherpa-tts.test.ts:133`: `eslint(no-new)` rejects constructing `SherpaOnnxTTS` only for side effects. Assigning the instance to a local variable (and optionally asserting it/config capture) preserves the test while satisfying the rule.

Evidence on the exact head:

- focused speech suite: **21 passed** across 4 files;
- full server `tsgo` typecheck: passed;
- Oxfmt on all 10 changed files: passed;
- Oxlint on the changed files: **2 errors**, exactly the two above, with 0 warnings.

I did not find another changed-file lint failure in this pass.

**Comment by LauraGPT** [comment]
I opened https://github.com/shouwangzhe/paseo/pull/2 against the current `feat/chinese-speech-models` branch to clear the two Oxlint blockers from the latest review:

- move offline recognizer model-config construction out of the constructor, reducing its reported complexity
- explicitly use the multilingual Kokoro TTS instance in the test, clearing `eslint(no-new)`

Fresh validation on top of `d3b1bdae`:

- 4 focused files / 21 tests passed
- full `@getpaseo/server` typecheck passed
- Oxfmt passed across all 10 files changed by this PR
- Oxlint passed across all 9 lintable changed files with 0 warnings and 0 errors
- `git diff --check` passed

The companion remains a draft for the branch owner to review and merge.

**Review by LauraGPT — APPROVED** [review]
Verified current head `3254baf2` after the Sherpa lint follow-up. The two blockers from my prior review are cleared.

- 4 focused speech files / 21 tests passed
- full `@getpaseo/server` typecheck passed
- Oxfmt passed across all 10 changed files
- Oxlint passed across all 9 lintable changed files with 0 warnings and 0 errors
- `git diff --check` passed

The extracted model-config builder preserves the Nemo, SenseVoice, and Paraformer native configuration values, and the Kokoro test now uses the constructed instance without changing runtime behavior.

**Comment by LauraGPT** [comment]
I completed a real SenseVoice CPU validation on exact PR head `3254baf23ec33af7dcf669f042cc4b5f191dbea2`.

Environment:

- Linux x86_64, glibc 2.39
- Node.js 22.20.0
- `sherpa-onnx-node` 1.12.28

The PR's downloader fetched, extracted, and verified `sense-voice-zh-en-ja-ko-yue-int8` in 18 seconds:

- `model.int8.onnx`: 239,233,841 bytes, SHA-256 `c71f0ce00bec95b07744e116345e33d8cbbe08cef896382cf907bf4b51a2cd51`
- `tokens.txt`: 315,894 bytes, SHA-256 `f449eb28dc567533d7fa59be34e2abca8784f771850c78a47fb731a31429a1dc`

Real worker-process inference results:

- FunASR official 16 kHz mono sample, SHA-256 `a1bd32dc78493c123f9625a66deee562aed2895f53fbc39f2cca3be7e6f4f20f`
  - `欢迎大家来体验达摩院推出的语音识别模型。`
- Bundled SenseVoice `zh.wav`, SHA-256 `b77f1794fe374a0ba1ee1dc458bfaf9349496cbbfc32780c50ba3c5a7ad8e373`
  - `开放时间早上9点至下午5点。`

There were no mocks or skipped samples, and the local speech workers exited cleanly.

One transparent boundary: the standalone transcription CLI on current `main` crashes before inference because its `RequestedSpeechProviders` object is missing `voiceTurnDetection`. I applied only the one-line setup fix from #2122 for these runs. None of this PR's 10 changed files were modified, so the model catalog, download/extraction, SenseVoice recognizer config, native worker dispatch, WAV decoding, and transcript generation all came from the exact PR head.

Commands used:

```bash
npm run speech:download --workspace=@getpaseo/server -- \
  --model sense-voice-zh-en-ja-ko-yue-int8 --models-dir "$MODELS_DIR"

npm run speech:transcribe:local --workspace=@getpaseo/server -- \
  "$WAV" --model sense-voice-zh-en-ja-ko-yue-int8 \
  --models-dir "$MODELS_DIR"
```

**Comment by shouwangzhe** [comment]
@boudra All review feedback on this PR has been addressed and the latest commit is approved with checks passing. It still shows as blocked for me due to merge permissions.

Could you merge it into `main` when you get a chance, or let me know if anything else is needed?

**Comment by boudra** [comment]
Closing this feature PR. Please discuss the workflow and shared need in [Discussions](https://github.com/getpaseo/paseo/discussions), following [CONTRIBUTING.md](https://github.com/getpaseo/paseo/blob/main/CONTRIBUTING.md).

**Comment by shouwangzhe** [comment]
邮件已收到，谢谢！
